pkg/ebpf/telemetry/errors_collector_linux.go

```diff
@@ -31,61 +31,70 @@ type EBPFErrorsCollector struct {
 	T                *EBPFTelemetry
 	ebpfMapOpsErrors *prometheus.Desc
 	ebpfHelperErrors *prometheus.Desc
+	supported        bool
 }
 
 // NewEBPFErrorsCollector initializes a new Collector object for ebpf helper and map operations errors
-func NewEBPFErrorsCollector() prometheus.Collector {
-	if supported, _ := ebpfTelemetrySupported(); !supported {
+func NewEBPFErrorsCollector(bpfDir string) prometheus.Collector {
+	var supported bool
+	var err error
+
+	if supported, err = ebpfTelemetrySupported(); err != nil {
 		return nil
 	}
 	return &EBPFErrorsCollector{
-		T:                newEBPFTelemetry(),
+		T:                initEBPTelemetry(bpfDir),
 		ebpfMapOpsErrors: prometheus.NewDesc(fmt.Sprintf("%s__errors", ebpfMapTelemetryNS), "Failures of map operations for a specific ebpf map reported per error.", []string{"map_name", "error"}, nil),
 		ebpfHelperErrors: prometheus.NewDesc(fmt.Sprintf("%s__errors", ebpfHelperTelemetryNS), "Failures of bpf helper operations reported per helper per error for each probe.", []string{"helper", "probe_name", "error"}, nil),
+		supported:        supported,
```

Comment:
Why create the object to do nothing, if not supported?

Reply:
This was done because we needed EBPF telemetry initialized even when telemetry is not supported, so we can patch the bytecode.

Removing this field, and initializing the EBPFTelemetry separately now.